test/cloudtest/test_k8s_node_recovery.py

```diff
@@ -0,0 +1,387 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+import subprocess
+import time
+from dataclasses import dataclass
+from textwrap import dedent
+
+import pytest
+
+from materialize import ui
+from materialize.cloudtest.app.materialize_application import MaterializeApplication
+
+
+@dataclass
+class ReplicaDefinition:
+    cluster_name: str
+    index: int
+    availability_zone: str
+
+    def get_name(self) -> str:
+        return f"{self.cluster_name}_r_{self.index}"
+
+
+@dataclass
+class ClusterDefinition:
+    name: str
+    replica_definitions: list[ReplicaDefinition]
+
+    def create_replica_definitions_sql(self) -> str:
+        replica_definitions = []
+        for replica in self.replica_definitions:
+            replica_definitions.append(
+                f"{replica.get_name()} (SIZE = '1', AVAILABILITY ZONE '{replica.availability_zone}')"
+            )
+
+        return ", ".join(replica_definitions)
+
+
+def default_compute_cluster() -> ClusterDefinition:
+    compute_cluster = ClusterDefinition("c_compute", [])
+    compute_cluster.replica_definitions.append(
+        ReplicaDefinition(compute_cluster.name, index=1, availability_zone="1")
+    )
+    return compute_cluster
+
+
+def default_storage_cluster() -> ClusterDefinition:
+    storage_cluster = ClusterDefinition("c_storage", [])
+    storage_cluster.replica_definitions.append(
+        ReplicaDefinition(storage_cluster.name, index=1, availability_zone="1")
+    )
+    return storage_cluster
+
+
+REQUIRED_PROGRESS = 5
+
+
+def populate(
+    mz: MaterializeApplication,
+    compute_cluster: ClusterDefinition,
+    storage_cluster: ClusterDefinition,
+) -> None:
+    all_clusters = [compute_cluster, storage_cluster]
+
+    drop_cluster_statements = [
```

Comment:
The dedents were messed up, the following patch seems to fix things for me:
```
diff --git a/test/cloudtest/test_k8s_node_recovery.py b/test/cloudtest/test_k8s_node_recovery.py
index ee37644a66..f67264bb1e 100644
--- a/test/cloudtest/test_k8s_node_recovery.py
+++ b/test/cloudtest/test_k8s_node_recovery.py
@@ -72,7 +72,7 @@ def populate(
         f"> DROP CLUSTER IF EXISTS {cluster.name};" for cluster in all_clusters
     ]
     # string needs the same indentation as testdrive script below
-    drop_cluster_statement_sql = "\n            ".join(drop_cluster_statements)
+    drop_cluster_statement_sql = "\n".join(drop_cluster_statements)
 
     create_cluster_statements = [
         f"> CREATE CLUSTER {cluster.name} REPLICAS ({cluster.create_replica_definitions_sql()});"
@@ -86,15 +86,14 @@ def populate(
             """
             > DROP MATERIALIZED VIEW IF EXISTS mv;
             > DROP SOURCE IF EXISTS source;
-
             """
-            + dedent(drop_cluster_statement_sql)
-            + "\n"
-            + dedent(create_cluster_statement_sql)
-            + "\n"
-            + dedent(
-                f"""
-
+        )
+        + dedent(drop_cluster_statement_sql)
+        + "\n"
+        + dedent(create_cluster_statement_sql)
+        + "\n"
+        + dedent(
+            f"""
             > SELECT name FROM mz_clusters WHERE name IN ('{compute_cluster.name}', '{storage_cluster.name}');
             {storage_cluster.name}
             {compute_cluster.name}
@@ -113,7 +112,6 @@ def populate(
             > SELECT COUNT(*) > 0 from mv;
             true
             """
-            )
         ),
         no_reset=True,
     )
```

Note the change in the placement of the closing brackets.